capstone/src/main/java/com/google/step/servlets/LoginServlet.java

```diff
@@ -0,0 +1,113 @@
+package com.google.step.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+
+import com.google.gson.Gson;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * Servlet that will handle login information.
+ */
+
+@WebServlet("/login")
+public class LoginServlet extends HttpServlet {
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    response.setContentType("application/json;");
+
+    UserService userService = UserServiceFactory.getUserService();
+    Map<String, Object> login = new HashMap<>();
+
+    if (userService.isUserLoggedIn()) {
+      String useremail = userService.getCurrentUser().getEmail();
+      String type = checkIfRestaurantOrUser(useremail);
+
+      // If user is loggedin but not in our system.
+      if (type.equals("None")) {
+        response.setStatus(HttpServletResponse.SC_FORBIDDEN);
+        return;
+      }
+
+      String urltoRedirectToAfterUserLogsOut = "/";
+      String logoutUrl = userService.createLogoutURL(urltoRedirectToAfterUserLogsOut);
+
+      login.put("Loggedin", true);
+      login.put("Type", type);
```

Comment:
This (the whole "checkIfRestaurantOrUser returns a string that gets passed through as-is to other functionality" thing) is an example of a place where you have a coupling issue. You have a method that you can use to determine whether an entity is a restaurant or a user, but then its return value is a string that gets returned to the frontend as-is. Then the frontend somehow depends on this thing.

It's fine if it ultimately becomes part of the URL, but at the very least, a cleaner way to model such things is to declare an `enum`, and optionally give that enum a method to retrieve "some url string" or whatever gets passed out. Then, you're not coupling "check if it's a user or a restaurant" and "the string we need to use in a URL".

The Effective Java book has a whole section on enums, give it a quick look.

Also, Megan's previous comment about organization of what servlets exists supersedes this comment. :)

Reply:
In the end this was changed as well due to the change in user-restaurant interaction.